wandb/beta/workflows.py

```diff
@@ -0,0 +1,266 @@
+) -> ArtifactInterface:
+    """Creates an artifact, populates it, and logs it with a run.
+    If a run is not present, we create one.
+
+    Args:
+        name: `str` - name of the artifact. If not scoped to a project, name will be suffixed by "-{run_id}".
+        type: `str` - type of the artifact, used in the UI to group artifacts of the same type.
+        entries: `Dict` - dictionary containing the named objects we want added to this artifact.
+        description: `str` - text description of artifact.
+        metadata: `Dict` - users can pass in artifact-specific metadata here, will be visible in the UI.
+        project: `str` - project under which to place this artifact.
+        scope_project: `bool` - if True, we will not suffix `name` with "-{run_id}".
+        job_type: `str` - Only applied if run is not present and we create one.
+            Used to identify runs of a certain job type, i.e "evaluation".
+
+    Returns:
+        ArtifactInterface
+
+    """
+    if wandb.run is None:
+        run = wandb.init(
+            project=project, job_type=job_type, settings=wandb.Settings(silent="true")
+        )
+    else:
+        run = wandb.run
+
+    if not scope_project:
+        name = f"{name}-{run.id}"
+
+    if metadata is None:
+        metadata = {}
+
+    art = wandb.Artifact(name, type, description, metadata, False, None)
+
+    for path in entries:
+        _add_any(art, entries[path], path)
+
+    # "latest" should always be present as an alias
+    aliases = wandb.util._resolve_aliases(aliases)
+    run.log_artifact(art, aliases=aliases)
+
+    return art
+
+
+def log_model(
+    model_obj: Any,
+    name: str = "model",
+    aliases: Optional[Union[str, List[str]]] = None,
+    description: Optional[str] = None,
+    metadata: Optional[dict] = None,
+    project: Optional[str] = None,
+    scope_project: Optional[bool] = None,
+    **kwargs: Dict[str, Any],
+) -> "_SavedModel":
+    """Logs a model object to enable model-centric workflows in the UI.
+    Supported frameworks include PyTorch, Keras, Tensorflow, Scikit-learn, etc.
+    Under the hood, we create a model artifact, bind it to the run that produced this model,
+    associate it with the latest metrics logged with `wandb.log(...)` and more.
+
+    Args:
+        model_obj: any model object created with the following ML frameworks: PyTorch, Keras, Tensorflow, Scikit-learn.
+        name: `str` - name of the model artifact that will be created to house this model_obj.
+        aliases: `str, List[str]` - optional alias(es) that will be applied on this model and allow for unique
+            identification. The alias "latest" will always be applied to the latest version of a model.
+        description: `str` - text description/notes about the model - will be visible in the Model Card UI.
+        metadata: `Dict` - model-specific metadata goes here - will be visible the UI.
+        project: `str` - project under which to place this artifact.
```

Comment:
oh so it could also belong to a run and a different project from the run's project?

Reply:
ah, `project` is only used when creating a run implicitly on the user's behalf